### DISCUSSION TITLE
Testing login: session data

### DISCUSSION THREAD

**Discussion by aidalgol** [keathley/wallaby#604] [Q&A] [answered]
I am trying to write a test for my application's login page.  I think I have most of it figured out, but I am stuck on checking the actual session data.  How can I assert on session data?

**Comment by mhanberg** [comment, marked as answer]
Assuming you are using Phoenix or Plug, you're session cookie is most likely encoded and/or encrypted. You could figure out some way using the Plug.Crypto module to decode it, but that wouldn't seem worth it to me.

What I would do is log in and assert that something on the page is specific to that user, like a name.

Or, if your auth solution keeps track of sessions in your database you could log in and then assert that a session record is stored in the database.

**Reply by aidalgol** [reply]
I was wondering whether I should test by looking at the UI, but I was also wary of being too purist about only testing the UI, and thought maybe I should look at the session data first, but when you lay it out like that, testing via the session data sounds overall worse.  And I realise now that I would be testing an implementation detail, rather than the functionality.
